projects/storefrontapp-e2e-cypress/cypress/integration/regression/product-carousel/product-carousel.e2e-spec.ts

```diff
@@ -0,0 +1,39 @@
+import * as cart from '../../../helpers/cart';
+import { viewportContext } from '../../../helpers/viewport-context';
+
+context('Product carousel', () => {
+  viewportContext(['mobile', 'desktop'], () => {
+    before(() => {
+      // Configure ProductCarouselComponent to render Add To Cart buttons for each slide.
+      cy.cxConfig({
+        cmsComponents: {
+          ProductCarouselComponent: {
+            data: {
+              composition: {
+                inner: ['ProductAddToCartComponent'],
+              },
+            },
+          },
+        },
+      });
+
+      cy.window().then((win) => win.sessionStorage.clear());
+      cy.visit('/');
+    });
+
+    describe('Add to Cart', () => {
+      it('should add a product to cart', () => {
+        const addToCart = cy
+          .get('cx-product-carousel cx-product-carousel-item cx-add-to-cart')
+          .first();
+        addToCart.should('be.visible');
+
+        const addToCartButton = addToCart.find('button');
+
+        addToCartButton.click({ force: true });
+
+        cart.checkAddedToCartDialog();
+      });
+    });
+  });
+});
```

Comment:
Ideally, we should test this on the CI before merging and the non-core tests don't run for feature branches.
One way I can think of is temporarily renaming the test file to `.core-e2e-spec.ts` so it runs for the feature branch PR.  Then rename it back after assessing the success of the test.  On the long run, we should find an easier way though.

Reply:
Sure, I can do that.